R/contingencytables.R

```diff
@@ -873,13 +923,51 @@ ContingencyTables <- function(jaspResults, dataset, options, ...) {
           row               <- c(row, total.proportions)
         }
 
+        if (options$residualsUnstandardized) {
+
+          row.unstandardized.residuals[["type[unstandardized.residuals]"]] <- gettext("Unstandardized residuals")
+          unstandardized.residuals                                         <- as.list(unstandardized.residuals.matrix[j,])
+          names(unstandardized.residuals)                                  <- paste0(names(unstandardized.residuals),"[unstandardized.residuals]")
+
+          unstandardized.residuals[["total[unstandardized.residuals]"]]    <- NULL
+
+          unstandardized.residuals                                         <- c(row.unstandardized.residuals, unstandardized.residuals)
+          row                                                              <- c(row, unstandardized.residuals)
+        }
+
+        if (options$residualsPearson) {
+
+          row.pearson.residuals[["type[pearson.residuals]"]]               <- gettext("Pearson residuals")
+          pearson.residuals                                                <- as.list(pearson.residuals.matrix[j,])
+          names(pearson.residuals)                                         <- paste0(names(pearson.residuals),"[pearson.residuals]")
+
+          pearson.residuals[["total[pearson.residuals]"]]                  <- NULL
+
+          pearson.residuals                                                <- c(row.pearson.residuals, pearson.residuals)
+          row                                                              <- c(row, pearson.residuals)
+        }
+
+        if (options$residualsStandardized) {
+
+          row.standardized.residuals[["type[standardized.residuals]"]]     <- gettext("Standardized residuals")
+          standardized.residuals                                           <- as.list(standardized.residuals.matrix[j,])
+          names(standardized.residuals)                                    <- paste0(names(standardized.residuals),"[standardized.residuals]")
+
+          standardized.residuals[["total[standardized.residuals]"]]        <- NULL
```

Comment:
Here is quite a lot of code duplication... perhaps you could make a function and reuse it three times